HPCC-22013 Replace website-antlr3 submodule with libantlr3c

Michael-Gardner opened the pull request:
Signed-off-by: Michael Gardner <michael.gardner@lexisnexisrisk.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

Commit:
HPCC-22013 Replace website-antlr3 submodule with libantlr3c

Signed-off-by: Michael Gardner <michael.gardner@lexisnexisrisk.com>

HPCCSmoketest commented:
Process of PR-12550, label: HPCC-22013 starts now.
The reason of this test is: Did not build yet.
Commit ID: 01035a6a9686f28ca83cab75d09a927e7ab047ee
Estimated completion time is ~0.5 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 01035a6a9686f28ca83cab75d09a927e7ab047ee
Build: success
Build: success
Install HPCC Platform
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |113 |113 |0 |0 |0 |36 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |26 sec  (00:00:26) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |17 sec  (00:00:17) |
|test (hthor) |829 |829 |0 |161 sec  (00:02:41) |
|test (thor) |754 |754 |0 |604 sec  (00:10:04) |
|test (roxie) |905 |905 |0 |221 sec  (00:03:41) |


HPCC Stop: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   47 sec (00:00:47) |  211 sec (00:03:31) |    0 sec (00:00:00) |    3 sec (00:00:03) |   19 sec (00:00:19) | 1312 sec (00:21:52) |   20 sec (00:00:20) | 1612 sec (00:26:52) |

Michael-Gardner commented:
@mckellyln Please review

mckellyln commented on esp/services/ws_sql/antlr3c.cmake:
```diff
@@ -41,7 +41,7 @@ endif()
 ExternalProject_Add(
     antlr3c
-    DOWNLOAD_COMMAND tar xzf ${CMAKE_CURRENT_SOURCE_DIR}/website-antlr3/download/C/libantlr3c-3.4.tar.gz
+    DOWNLOAD_COMMAND tar xzf ${CMAKE_CURRENT_SOURCE_DIR}/libantlr3c/libantlr3c-3.4.tar.gz
```
It is a little convoluted that the build untars a tarfile with the same files as what is already in the source.  But this is not really any different than it was.
The:
build/esp/services/ws_sql/antlr3c/src/libantlr3c-3.4/src
dir created from the tar xf cmd, and the:
HPCC-Platform/esp/services/ws_sql/libantlr3c/src
are identical.  Same for include dir and m4 dir.

mckellyln commented:
@Michael-Gardner I am ok with this change.  It is better than before by having the submodule only have the files we use, and it works correctly.  But I think it may be worth another look, as it seems the submodule source from the tarfile appears to match the files already in the source dir.

richardkchapman commented:
I think there's a reason we do it this way (but I can't remember what it is!)

mckellyln commented:
Perhaps then we don't need the source in the source dir ?
Something to look at, but as a separate issue ?
But I am good with this PR as is, it is good and an improvement over the previous.

Michael-Gardner commented:
@mckellyln Originally I had thought I needed to include the source as is, with the license file, but I also wanted to include the archive to retain the same logic that we were using in the ExternalProject_Add's DOWNLOAD_COMMAND.  I could probably just change the DOWNLOAD_COMMAND to cp that entire directory over into the build directory, and then remove the archive.  I'll do that tomorrow and get back to you.

mckellyln commented:
@Michael-Gardner back to you one more time.

Commit:
HPCC-22013 Remove tar and change download step to copy instead of extract

Signed-off-by: Michael Gardner <michael.garder@lexisnexisrisk.com>

HPCCSmoketest commented:
Process of PR-12550, label: HPCC-22013 starts now.
The reason of this test is: New commit.
Commit ID: 736a74082c904edaca312c10fbfbb66cbe4a2bd8
Estimated completion time is ~0.5 hour(s)

Michael-Gardner commented:
@mckellyln I modified the libantlr3c repository and removed that redundant jar.  I altered the DOWNLOAD step on add external project to do a copy instead of an extraction.  Had to make other small changes to insure that it worked correctly, and could find all the necessary files, but nothing major.

richardkchapman commented:
7.4 would be a more appropriate target

Michael-Gardner commented:
Rebasing it now

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 736a74082c904edaca312c10fbfbb66cbe4a2bd8
Build: success
Build: success
Install HPCC Platform
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |113 |113 |0 |0 |0 |35 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |24 sec  (00:00:24) |
|setup (thor) |11 |11 |0 |43 sec  (00:00:43) |
|setup (roxie) |11 |11 |0 |24 sec  (00:00:24) |
|test (hthor) |829 |829 |0 |184 sec  (00:03:04) |
|test (thor) |754 |754 |0 |598 sec  (00:09:58) |
|test (roxie) |905 |905 |0 |221 sec  (00:03:41) |


HPCC Stop: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   32 sec (00:00:32) |  209 sec (00:03:29) |    0 sec (00:00:00) |    2 sec (00:00:02) |   19 sec (00:00:19) | 1314 sec (00:21:54) |   19 sec (00:00:19) | 1595 sec (00:26:35) |

mckellyln commented:
I see it has already been merged.
